cli/tests/test_tag_command.rs

```diff
@@ -117,3 +117,92 @@ fn test_tag_list() {
     added_targets: commit2
     "###);
 }
+
+#[test]
+fn test_tag_patterns() {
```

Comment:
nit: can you add tests to lib/tests/test_revset.rs instead? Library tests is much faster than CLI tests, and I don't think e2e CLI test is needed for this change.

I think you can copy setup code from `test_resolve_symbol_tags()` and `test_evaluate_expression_bookmarks()`. There was no library test for `tags()` expression yet.

Reply:
Have added these. The bookmark tests seem fairly exhaustive, so just copied what they had.